ChaosMod/Effects/db/Player/PlayerFakeDeath.cpp

```diff
@@ -141,8 +159,23 @@ static void OnStart()
 							{
 								SET_VEHICLE_DOOR_BROKEN(veh, i, false);
 							}
-							Vector3 coords = GET_ENTITY_COORDS(veh, false);
-							ADD_EXPLOSION(coords.x, coords.y, coords.z, 7, 999, true, false, 1, true);
+							Vector3 vehCoords = GET_ENTITY_COORDS(veh, false);
+							Vector3 plrCoords = GET_ENTITY_COORDS(playerPed, false);
+							ADD_EXPLOSION(vehCoords.x, vehCoords.y, vehCoords.z, 7, 999, true, false, 1, true);
+
+							//skip right to the cleanup if the player bailed out of the vehicle and is far enough away.
+							if (!IS_PED_IN_VEHICLE(playerPed, veh, false))
+							{
+								if (GET_DISTANCE_BETWEEN_COORDS(vehCoords.x, vehCoords.y, vehCoords.z, plrCoords.x, plrCoords.y, plrCoords.z, false) > 1.5)
+								{
+									GetComponent<EffectDispatcher>()->OverrideEffectNameId("player_fakedeath",
+									                                                       fakeEffectId);
+									WAIT(2000);
+
+									currentMode = FakeDeathState::cleanup;
+									goto SKIP_TO_CLEANUP; //eww, a goto statement
```

Comment:
eww, a goto statement

Reply:
What you want me to do about it, goto is the new meta

Reply:
goto is ugly, errorprone and not something you should use :)

Reply:
What you want me to use?

Reply:
I’ll fix this when I can.

Reply:
All fixed 👍